src/main/java/Pawn.java

```diff
@@ -54,35 +59,28 @@ public long attack_coverage(int side, LocationBitboard board) {
                 | board.whiteBishop[0] | board.whiteQueen[0] | board.whiteKing[0];
         long white_board = board.blackPawn[0] | board.blackRook[0] | board.blackKnight[0]
                 | board.blackBishop[0] | board.blackQueen[0] | board.blackKing[0];
+        long white_pawn_attacks = 0L;
+        long black_pawn_attacks = 0L;
+        for (int i = 0; i < 64; i++) {
+            long pawn_pos = 1L >>> i;
+            if ((pawn_pos & board.whitePawn[0]) != 0L) {
+                white_pawn_attacks |= pawn_pos << 7;
+                white_pawn_attacks |= pawn_pos << 9;
+            } else if ((pawn_pos & board.blackPawn[0]) != 0L) {
+                black_pawn_attacks |= pawn_pos >>> 7;
+                black_pawn_attacks |= pawn_pos >>> 9;
+            }
+        }
 
         // start with no attack positions
-        long pawn_attacks = 0L;
-        // We will use 1L and shift the 1 right once each time to check all board positions
-        long position_checker = 1L;
-
+        long valid_pawn_attacks = 0L;
```

Comment:
Looks like you cleaned up the code too to remove unnecessary variables, thank you.